modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/Slot.java

```diff
@@ -52,6 +61,16 @@
     private boolean isSlotActive;
 
     /**
+     * Indicates whether the slot is a fresh one or an overlapped one
+     */
+    private boolean isAnOverlappingSlot;
```

Comment:
Yes this is for the convenience factor. The Slotstates are maintained within the object as a list, and to refer the list to find out the overlapping slots would be unnecessary. the states are maintained to validate the transition from one stage to the other.